### DISCUSSION TITLE
21.11.2023 Area A Data Modeling

### DISCUSSION THREAD

**Discussion by hampusnasstrom** [FAIRmat-NFDI/AreaA-data_modeling_and_schemas#78] [Data Modeling Meeting]
## Agenda
We will discuss the `System` base section.

## Meeting Notes
Attendees: @hampusnasstrom, @JFRudzinski, @mkuehbach, @budschi, @JosePizarro3, @aalbino2, @ka-sarthak, @ndaelman-hu 

- We discussed the need for a best-practice on plugin writing
  - Markus Scheidgen wrote some docs on resuable code
  - We will deal with this in next weeks meeting
- Chema introduces the current state of the `System` remodelling
  - [Google slides](https://docs.google.com/presentation/d/1fjgRpjE_u90S_ddpp-IJ8A9Z2ScZRav_nMMARwFNGc4/edit?usp=sharing)
  - We should work on an example basis, trying to implement them and see what needs to change
- Discussion on the `System` section
  - Currently we have all parameters in the *base* section but this should be specialized later
  - Chema made the point that over-nesting can cause problems when trying to generalize
  - Sebastian raised the point that very specialized sections (inheriting in the end from `System`) should live in plugins
  - We clarified that the `System` in the google slides actually refers to a subtype of `System`
  - Nesting `System` helps with time evolution where we can assign a time step
  - How do we deal with specifying that certain `System`s should be present in a specific use case

## Tasks

- [ ] Area A (@hampusnasstrom) will add a system description for the solar cell example and some additional ones
- [ ] Review the examples within 3 weeks (until 11.12.2023)

**Comment by JFRudzinski** [comment]
![Screenshot from 2023-11-22 10-52-01](https://github.com/FAIRmat-NFDI/AreaA-data_modeling_and_schemas/assets/11678418/c2d8904f-a03a-4c91-90d7-7dd3a651c725)

**Comment by JFRudzinski** [comment]
I just wanted to add a comment that I thought about during the meeting but didn't want to cloud the discussion:

I **REALLY** think we need to stop using the term "topology", so that it does not continue to leak into our schema names and such accidentally. This is my fault because it's the common name for the system hierarchy in my community, but I think more generally it is a bad term since it is not obvious what it means and is even misleading depending on your background. I will put forth "System groups" as an initial suggest, but I am certainly open to alternatives.

**Reply by JosePizarro3** [reply]
_Grouping_ sounds good, and it is indeed what we should be using to refer the System nested in itself. It is just that "topology tree" works good; but, how would you call this tree otherwise? "System grouping" directly?

**Reply by mkuehbach** [reply]
On the naming of the symbol "topology" aka "grouping":
 PLACEHOLDER_set is what I typically use, i.e. 
 triangle_set, hexahedron_set, volume_element_set, system_set

On the use of the symbol "topology":
Indeed I agree it should be used only if one really talks about a concept that talks about the mathematical term and graph that is behind the topology of something

Why I will from now use the established terminology from the respective semantic web technology fields of research:
because the least what nomad needs is yet more jargon which does nothing but reimplement the wheel
